applications/FSIapplication/custom_utilities/variable_redistribution_utility.cpp

```diff
@@ -92,32 +98,38 @@ void VariableRedistributionUtility::CallSpecializedDistributePointValues(
     double Tolerance,
     unsigned int MaximumIterations)
 {
-    // This function only dispatches the call to the correct specialization
-    Geometry< Node<3> >& rReferenceGeometry = rModelPart.ConditionsBegin()->GetGeometry();
-    const GeometryData::KratosGeometryFamily GeometryFamily = rReferenceGeometry.GetGeometryFamily();
-    const unsigned int PointsNumber = rReferenceGeometry.PointsNumber();
+    // Check if there is any condition in the current partition
+    int n_conds = rModelPart.GetCommunicator().LocalMesh().NumberOfConditions();
+    rModelPart.GetCommunicator().SumAll(n_conds);
 
-    if (GeometryFamily == GeometryData::Kratos_Linear && PointsNumber == 2)
-    {
-        VariableRedistributionUtility::SpecializedDistributePointValues<GeometryData::Kratos_Linear,2,TValueType>(
-            rModelPart,
-            rPointVariable,
-            rDistributedVariable,
-            Tolerance,
-            MaximumIterations);
-    }
-    else if (GeometryFamily == GeometryData::Kratos_Triangle && PointsNumber == 3)
-    {
-        VariableRedistributionUtility::SpecializedDistributePointValues<GeometryData::Kratos_Triangle,3,TValueType>(
-            rModelPart,
-            rPointVariable,
-            rDistributedVariable,
-            Tolerance,
-            MaximumIterations);
-    }
-    else
-    {
-        KRATOS_ERROR << "Unsupported geometry type with " << PointsNumber << " points." << std::endl;
+    // If there is conditions, this function dispatches the call to the correct specialization
+    if (n_conds != 0){
+        Geometry< Node<3> >& rReferenceGeometry = rModelPart.ConditionsBegin()->GetGeometry();
+        const GeometryData::KratosGeometryFamily GeometryFamily = rReferenceGeometry.GetGeometryFamily();
+        const unsigned int PointsNumber = rReferenceGeometry.PointsNumber();
```

Comment:
Same as above, one given partition might not have conditions and these calls will fail.

Reply:
In principle this is fixed with the new "dummy" methods